elastic/beats
# Add new config options, and add workaround for older Ubuntu systemd reporting to system/service

## What does this PR do?

This fixes a rather odd bug in the version of systemd that ships with older Ubuntu releases, namely that instantiated units no longer report themselves once the process is dead. This adds a `service.track_instantiated` flag, among others, that'll "track" instantiated processes until a given timeout period.

## Why is it important?

This is a workaround for a bug in an Ubuntu LTS release that we currently support.

## Checklist


## How to test this PR locally

- Pull down PR.
- On your beats development machine, run `vagrant up ubuntu1604` to get the right VM running. Or just run this natively on Ubuntu 16.04.
- Create a file `/lib/systemd/system/tester@.service` with the following:
```
[Unit]
Description=Instance tester time %i

[Service]
ExecStart=-/bin/sleep %i
KillMode=mixed
Delegate=yes
TasksMax=infinity
```
- Use this for your metricbeat module config:
```
- module: system
  period: 10s
  metricsets:
    - service

  service.pattern_filter: ["tester*"]
  service.track_instantiated: true
  service.track_instantiated_timeout: 60s
```
- run `systemctl start tester@60.service`
- Start metricbeat right away after.
- After 60 seconds, the `tester@60` service will die
- Make sure that metricbeat is still reporting the service, with `state: inactive`
